src/daft-dsl/src/expr.rs

```diff
@@ -265,28 +273,28 @@ impl AggExpr {
                     },
                 ))
             }
-            ApproxPercentile(ApproxPercentileParams {
-                child: expr,
-                percentiles,
+            &ApproxPercentile(ApproxPercentileParams {
+                child: ref expr,
+                ref percentiles,
                 force_list_output,
             }) => {
                 let field = expr.to_field(schema)?;
-                Ok(Field::new(
-                    field.name.as_str(),
-                    match &field.dtype {
-                        dt if dt.is_numeric() => if percentiles.len() > 1 || *force_list_output {
-                            DataType::FixedSizeList(Box::new(DataType::Float64), percentiles.len())
-                        } else {
-                            DataType::Float64
-                        },
-                        other => {
-                            return Err(DaftError::TypeError(format!(
-                                "Expected input to approx_percentiles() to be numeric but received dtype {} for column \"{}\"",
-                                other, field.name,
-                            )))
-                        }
-                    },
-                ))
+                let dtype = field.dtype.assert_is_numeric().map(|()| {
+                    if percentiles.len() > 1 || force_list_output {
+                        DataType::FixedSizeList(Box::new(DataType::Float64), percentiles.len())
+                    } else {
+                        DataType::Float64
+                    }
+                })?;
+                Ok(Field::new(field.name, dtype))
+            }
+            CountDistinct(expr) => {
+                let field = expr.to_field(schema)?;
+                let dtype = field
+                    .dtype
+                    .assert_is_numeric()
+                    .map(|()| DataType::Float64)?;
+                Ok(Field::new(field.name, dtype))
```

Comment:
Should a CountDistinct not be returning something more like a `u64` - when would we need a float for counts?